pkg/storage/command_queue.go

```diff
@@ -801,3 +801,27 @@ func (cq *CommandQueue) metrics() CommandQueueMetrics {
 		TreeSize:        int32(cq.treeSize()),
 	}
 }
+
+func (cq *CommandQueue) GetCommandQueueState() *CommandQueues {
+	return &CommandQueues{
+		Reads:  copyCommandsFromIntervalTree(cq.reads),
+		Writes: copyCommandsFromIntervalTree(cq.writes),
+	}
+}
+
+func copyCommandsFromIntervalTree(tree interval.Tree) []*CommandQueueCommand {
+	var result []*CommandQueueCommand
+	it := tree.Iterator()
+	for {
+		// TODO: is there a more idiomatic way to iterate?
```

Comment:
https://github.com/cockroachdb/cockroach/blob/bddd7c111881b9bdc5c5d3d5e965c223e50a8483/pkg/util/interval/interval.go#L175-L180 is probably what you're looking for.

Reply:
Perfect, thx